Lib/socket.py

```diff
@@ -728,6 +728,149 @@ def create_connection(address, timeout=_GLOBAL_DEFAULT_TIMEOUT,
     else:
         raise error("getaddrinfo returns an empty list")
 
+def supports_hybrid_ipv46():
+    """Return True if the platform supports creating a SOCK_STREAM socket
+    which can handle both AF_INET and AF_INET6 (IPv4 / IPv6) connections.
+    """
+    if not has_ipv6 \
+            or not hasattr(_socket, 'IPPROTO_IPV6') \
+            or not hasattr(_socket, 'IPV6_V6ONLY'):
+        return False
+    try:
+        with socket(AF_INET6, SOCK_STREAM) as sock:
+            sock.setsockopt(IPPROTO_IPV6, IPV6_V6ONLY, 0)
+            return True
+    except error:
+        return False
+
+def bind_socket(address, *, family=AF_UNSPEC, type=SOCK_STREAM, backlog=128,
```

Comment:
The `backlog` default value topic is a tricky one. Some words have already been spent here:
https://github.com/python/cpython/pull/11215#discussion_r242762410

> Would it make sense to use backlog=None here and call sock.listen() with no parameter if it's None, or sock.listen(backlog) otherwise?

Yes, I think it would probably  be OK.

> Maybe raise an exception if backlog is set but socktype != SOCK_STREAM?

Mmmm... I think that would be a bit too strict/punitive.